File: support_tools/swigwin-3.0.12/Examples/test-suite/python/li_cstring_runme.py
from li_cstring import *


if count("ab\0ab\0ab\0", 0) != 3:
    raise RuntimeError

if test1() != "Hello World":
    raise RuntimeError

if test2() != " !\"#$%&'()*+,-./0123456789:;<=>?@ABCDEFGHIJKLMNOPQRSTUVWXYZ[\]^_":
    raise RuntimeError

if test3("hello") != "hello-suffix":
    print test3("hello")
    raise RuntimeError

if test4("hello") != "hello-suffix":
    print test4("hello")
    raise RuntimeError

if test5(4) != 'xxxx':
    raise RuntimeError

if test6(10) != 'xxxxx':
    raise RuntimeError

if test7() != "Hello world!":
    raise RuntimeError

if test8() != " !\"#$%&'()*+,-./0123456789:;<=>?@ABCDEFGHIJKLMNOPQRSTUVWXYZ[\]^_":
    raise RuntimeError

File: support_tools/swigwin-3.0.12/Examples/test-suite/ruby/li_cstring_runme.rb
#!/usr/bin/env ruby
#
# Test for li_cstring.i
#

require 'swig_assert'
require 'li_cstring'

include Li_cstring

swig_assert_each_line <<EOF
count("hello", 'l'[0]) == 2
test1 == 'Hello World'
test2
test3('hello') == 'hello-suffix'
test4('hello') == 'hello-suffix'
test5(5) == 'xxxxx'
test6(6) == 'xxx'
test7    == 'Hello world!'
test8    == (32..32+63).map {|x| x.chr }.join
EOF

